modules/gcp/compute.go

```diff
@@ -171,13 +286,66 @@ func GetInstanceIdsForInstanceGroupE(t *testing.T, projectID string, zone string
 		}
 		return nil
 	}); err != nil {
-		return nil, fmt.Errorf("InstanceGroups.ListInstances(%s) got error: %v", groupName, err)
+		return nil, fmt.Errorf("InstanceGroups.ListInstances(%s) got error: %v", ig.Name, err)
 	}
 
 	return instanceIDs, nil
 }
 
-// NewComputeService creates a new Compute service.
+// GetRandomInstance returns a randomly selected Instance from the Regional Instance Group
+func (ig *RegionalInstanceGroup) GetRandomInstance(t *testing.T) *Instance {
+	instance, err := ig.GetRandomInstanceE(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return instance
+}
+
+// GetRandomInstanceE returns a randomly selected Instance from the Regional Instance Group
+func (ig *RegionalInstanceGroup) GetRandomInstanceE(t *testing.T) (*Instance, error) {
+	instanceIDs := ig.GetInstanceIds(t)
+	clusterSize := int(ig.Size)
+	randIndex := random.Random(1, clusterSize)
+
+	if randIndex > len(instanceIDs) {
+		return nil, fmt.Errorf("Could not find any instances in Regional Instance Group %s in Region %s", ig.Name, ig.Region)
+	}
+
+	instanceID := instanceIDs[randIndex-1]
+	instance := FetchInstance(t, ig.projectID, instanceID)
+
+	return instance, nil
+}
+
+// TODO: Is there some way to avoid the total duplication of this function and the Regional Instance Group equivalent in Go?
```

Comment:
Define a common interface for `ZonalInstanceGroup` and `RegionalInstanceGroup` and create a standalone method that takes in that interface as a parameter and returns a random `Instance`.

Reply:
Thanks, updated in 485ca9ad89b866e972ef841a2d915e4f9d2b9ce3.